### PR TITLE
feat(relayer): Introduce VaraArgs with retries count

### LINKED ISSUE #145: Use retry feature of gclient/gsdk to improve checkpoints relayer stability
### File Location(s)

_No response_

### Proposal

after #140 

Retry functionality added to gclient/gsdk here - https://github.com/gear-tech/gear/pull/4225 . It is disabled by default so checkpoints relayer has to be adjusted.

### PR DESCRIPTION
**Pull request by gshep**
Resolves #145 

@gear-tech/dev

### CODE REVIEW

**Comment by gshep** [comment]
We may also address the TODO to `VaraEndpointArgs` within that PR

**Review by mertwole — COMMENTED** [review]
Please also introduce it in `gear-rpc-client`

**Comment by mertwole** [comment]
> We may also address the TODO to `VaraEndpointArgs` within that PR

I think it fits great into this PR

File: relayer/src/main.rs [hunk 1]
```diff
@@ -382,22 +390,17 @@ async fn main() {
...
+        .retries(args.vara_rpc_retries)
+        .suri(&args.vara_suri)
+        .build(WSAddress::new(&args.vara_domain, args.vara_port))
         .await
         .expect("Failed to create gclient client")
 }
 
-async fn create_gear_client(args: &VaraEndpointArg) -> GearApi {
-    GearApi::new(&args.vara_endpoint)
+async fn create_gear_client(domain: &str, port: u16, retries: u8) -> GearApi {
```
Comment (mertwole) [inline relayer/src/main.rs:402]:
```suggestion
async fn create_gear_client(args: &VaraArgs) -> GearApi {
```

File: relayer/src/main.rs [hunk 2]
```diff
@@ -78,16 +77,26 @@ struct RelayMerkleRootsArgs {
...
+    /// Port of the VARA RPC endpoint
+    #[arg(long, default_value = "9944", env = "VARA_PORT")]
+    vara_port: u16,
+
+    /// Substrate URI that identifies a user by a mnemonic phrase or
+    /// provides default users from the keyring (e.g., "//Alice", "//Bob",
+    /// etc.). The password for URI should be specified in the same `suri`,
+    /// separated by the ':' char
+    #[arg(long, default_value = "//Alice", env = "VARA_SURI")]
+    vara_suri: String,
```
Comment (mertwole) [inline relayer/src/main.rs:95]: I think it should be in a separate struct as it's required not by every relayer

File: tools/genesis-config/src/main.rs [hunk 3]
```diff
@@ -29,6 +29,11 @@ struct FetchArgs {
     /// Port of the Gear RPC endpoint
     #[arg(long = "gear-port", default_value = "443", env = "GEAR_PORT")]
     gear_port: u16,
+
+    /// Set retries of the VARA RPC client
+    #[arg(long, default_value = "3", env = "GEAR_RPC_RETRIES")]
+    gear_rpc_retries: u8,
```
Comment (mertwole) [inline tools/genesis-config/src/main.rs:35]: I think somethig like `const GEAR_RPC_RETRIES = 3` will be ok for this use-case

**Review by mertwole — CHANGES_REQUESTED** [review]

File: relayer/src/main.rs [hunk 4]
```diff
@@ -51,7 +51,7 @@ enum CliCommands {
 #[derive(Args)]
 struct RelayMessagesArgs {
     #[clap(flatten)]
-    vara_endpoint: VaraEndpointArg,
+    vara: VaraArgs,
```
Comment (mertwole) [inline relayer/src/main.rs:54]:
```suggestion
    vara_args: VaraArgs,
```

File: relayer/src/main.rs [hunk 5]
```diff
@@ -67,7 +67,7 @@ struct RelayMessagesArgs {
 #[derive(Args)]
 struct RelayMerkleRootsArgs {
     #[clap(flatten)]
-    vara_endpoint: VaraEndpointArg,
+    vara: VaraArgs,
```
Comment (mertwole) [inline relayer/src/main.rs:70]:
```suggestion
    vara_args: VaraArgs,
```

**Review by mertwole — APPROVED** [review]

**Pull request merged**